velox/docs/configs.rst

```diff
@@ -436,6 +441,16 @@ Each query can override the config by setting corresponding query session proper
      - string
      - 10MB
      - Maximum bytes for sort writer in one batch of output. This is to limit the memory usage of sort writer.
+   * - file-preload-threshold
+     -
+     - integer
+     - 1MB
+     - Usually Velox fetches the meta data firstly then fetch the rest of file. But if the file is very small, Velox can fetch the whole file directly to avoid multiple IO requests. The parameter control the threshold when whole file is fetched. 
+   * - directory-size-guess
```

Comment:
> directory-size-guess

This name is a bit confusing. Which directory does this refer to? This is not a directory on the file system, is it? Wondering if we could come up with a better name here.

Reply:
According to the code, I don't see anything related to directory. Looks like it just read the following data after meta data. In dwrf, is it the directory data?

Reply:
I'm not sure. Seems like it was introduced in https://github.com/facebookincubator/velox/pull/3891 and is used in Parquet reader.

@zhenghaozzz Would you help clarify?

Reply:
This PR only made `DIRECTORY_SIZE_GUESS` configurable. I always wondered the meaning of this as well.

Reply:
@mbasmanova I agree this name is confusing but I still keep the name. It is used to guess the size of metadata in #3891.

Reply:
@zhenghaozzz Would you explain what this parameter is used for and help come up with a more descriptive name?

Reply:
Is this an estimate of how big the file footer is?

Reply:
`FOOTER_SIZE_GUESS` or `FOOTER_LENGTH_GUESS`. Whatever, it guesses the size of something called footer DwrfReader or ParquetReader.

Reply:
Wondering if this is somehow related to StripeDictionaryCache. An internal modification to DWRF "caches" stripe metadata next to file footer to allow for fetching it in a single IO together with the footer.

CC: @HuamengJiang